geode-core/src/distributedTest/java/org/apache/geode/internal/cache/tier/sockets/RedundancyLevelTestBase.java

```diff
@@ -132,380 +124,231 @@ public final void postSetUp() throws Exception {
+    assertThat(bs).isNotNull();
+    assertThat(bs.getAcceptor()).isNotNull();
+    assertThat(bs.getAcceptor().getCacheClientNotifier()).isNotNull();
 
-      CacheServerImpl bs = (CacheServerImpl) cache.getCacheServers().iterator().next();
-
-      assertNotNull(bs);
-      assertNotNull(bs.getAcceptor());
-      assertNotNull(bs.getAcceptor().getCacheClientNotifier());
-      // one client is connected to this server
-      final CacheClientNotifier ccn = bs.getAcceptor().getCacheClientNotifier();
-      wc = new WaitCriterion() {
-        String excuse;
-
-        public boolean done() {
-          return ccn.getClientProxies().size() == 1;
-        }
-
-        public String description() {
-          return excuse;
-        }
-      };
-      Wait.waitForCriterion(wc, 3 * 60 * 1000, 1000, true);
-    } catch (Exception ex) {
-      Assert.fail("exception in verifyCCP()", ex);
-    }
+    // no client is connected to this server
+    assertThat(bs.getAcceptor().getCacheClientNotifier().getClientProxies().size()).isEqualTo(0);
   }
 
-  public static void verifyInterestRegistration() {
-    try {
-      WaitCriterion wc = new WaitCriterion() {
-        public boolean done() {
-          return cache.getCacheServers().size() == 1;
-        }
+  static void verifyCCP() {
+    await().atMost(180, SECONDS).pollInterval(2, SECONDS)
+        .until(() -> cache.getCacheServers().size(), equalTo(1));
+    CacheServerImpl bs = (CacheServerImpl) cache.getCacheServers().iterator().next();
 
-        public String description() {
-          return "Number of bridge servers (" + cache.getCacheServers().size() + ") never became 1";
-        }
-      };
-      Wait.waitForCriterion(wc, 180 * 1000, 2000, true);
-
-      CacheServerImpl bs = (CacheServerImpl) cache.getCacheServers().iterator().next();
-      assertNotNull(bs);
-      assertNotNull(bs.getAcceptor());
-      assertNotNull(bs.getAcceptor().getCacheClientNotifier());
-      final CacheClientNotifier ccn = bs.getAcceptor().getCacheClientNotifier();
-      wc = new WaitCriterion() {
-        public boolean done() {
-          return ccn.getClientProxies().size() > 0;
-        }
+    assertThat(bs).isNotNull();
+    assertThat(bs.getAcceptor()).isNotNull();
+    assertThat(bs.getAcceptor().getCacheClientNotifier()).isNotNull();
 
-        public String description() {
-          return "Notifier's proxies is empty";
-        }
-      };
-      Wait.waitForCriterion(wc, 180 * 1000, 2000, true);
+    // one client is connected to this server
+    final CacheClientNotifier ccn = bs.getAcceptor().getCacheClientNotifier();
+    await().atMost(180, SECONDS).pollInterval(1, SECONDS).until(() -> ccn.getClientProxies().size(),
+        equalTo(1));
+  }
 
-      Iterator iter_prox = ccn.getClientProxies().iterator();
+  static void verifyInterestRegistration() {
+    await("Number of bridge servers (" + cache.getCacheServers().size() + ") never became 1")
```

Comment:
I would delete all usage of pollInterval and then crank up the atMost to FIVE MINUTES.

Reply:
sometimes, I don't want the system to poll more frequently than specified. Thats why I use pollInterval. Do you think that is a bad approach? I thought the default was 100ms